### PR TITLE
onlyKeeper for rebalance() and admin & keeper addresses separation

### PR DESCRIPTION
**Pull request by brkomir**

### CODE REVIEW

**Review by coolhill — APPROVED** [review]

File: src/steth/scWETH.sol [hunk 1]
```diff
@@ -80,7 +81,7 @@ contract scWETH is sc4626, IFlashLoanRecipient {
     }
 
     constructor(ConstructorParams memory _params)
-        sc4626(_params.admin, _params.weth, "Sandclock WETH Vault", "scWETH")
+        sc4626(_params.admin, _params.keeper, _params.weth, "Sandclock WETH Vault", "scWETH")
     {
         if (_params.admin == address(0)) revert ZeroAddress();
```
Comment (dxganta) [inline src/steth/scWETH.sol:86]:
```suggestion
        if (_params.admin == address(0)) revert ZeroAddress();
        if (_params.keeper== address(0)) revert ZeroAddress();
```

Comment (dxganta) [reply src/steth/scWETH.sol:86]: Just to be extra safe.

Comment (brkomir) [reply src/steth/scWETH.sol:86]: we should move these checks to sc4626.sol

Comment (brkomir) [reply src/steth/scWETH.sol]: fixed [here](https://github.com/lindy-labs/sandclock-contracts/pull/45/commits/83f382b67556acddcad66a82a3f487693d5af1ab)

**Review by dxganta — APPROVED** [review]

**Pull request merged**